### ISSUE TITLE
[New rule] Prefer future void

### ISSUE THREAD

**Issue by incendial** [dart-code-checker/dart-code-metrics#115]
**Please describe what the rule should do:**
Warns when return type is `Future` or `Future<Null>` instead of a `Future<void>`.

**What category of rule is this? (place an "X" next to just one item)**

[X] Warns about a potential error (problem)
[ ] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about (it will be better if you can provide both good and bad examples):**

Bad:
```dart
Future schedule(void Function callback) => _scheduler.add(callback);

Future<Null> (void Function callback) => _scheduler.add(callback);
```

Good:
```dart
Future<void> (void Function callback) => _scheduler.add(callback);
```

**Are you willing to submit a pull request to implement this rule?**
Yes

**Comment by dkrutskikh** [comment]
@incendial I found similar rule https://dart-lang.github.io/linter/lints/prefer_void_to_null.html
